packages/odyssey-react/src/components/Button/Button.module.scss

```diff
@@ -70,69 +71,73 @@
 .secondaryVariant {
   border-color: var(--ColorBorderSecondary);
   background-color: var(--ColorBackgroundSecondary);
-  color: $color-primary-base;
+  color: var(--ColorTextSecondary);
 
   &:hover {
     border-color: var(--ColorBorderSecondaryHover);
     background-color: var(--ColorBackgroundSecondaryHover);
-    color: $white;
+    color: var(--ColorTextSecondaryHover);
   }
 
   &:focus {
     border-color: var(--ColorBorderSecondaryFocus);
     background-color: var(--ColorBackgroundSecondaryFocus);
-    color: $color-primary-base;
+    color: var(--ColorTextSecondaryFocus);
   }
 
   &:disabled {
     border-color: var(--ColorBorderSecondaryDisabled);
     background-color: var(--ColorBackgroundSecondaryDisabled);
-    color: $color-primary-light;
+    color: var(--ColorTextSecondaryDisabled);
   }
 }
 
 .dangerVariant {
-  background-color: $color-danger-base;
+  background-color: var(--ColorBackgroundDanger);
 
   &:hover {
-    border-color: $color-danger-dark;
-    background-color: $color-danger-dark;
+    border-color: var(--ColorBorderDangerHover);
+    background-color: var(--ColorBackgroundDangerHover);
   }
 
   &:focus {
-    @include outline($focus-ring-danger);
+    @include outline(var(--ColorFocusDanger), var(--FocusOutlineWidth));
 
-    background-color: $color-danger-dark;
+    background-color: var(--ColorBackgroundDangerFocus);
   }
 
   &:disabled {
-    border-color: $color-danger-light;
-    background-color: $color-danger-light;
+    border-color: var(--ColorBorderDangerDisabled);
+    background-color: var(--ColorBackgroundDangerDisabled);
   }
 }
 
 .dismissVariant,
 .dismissInvertedVariant {
-  padding-block: ($spacing-xs-em * 0.5);
-  padding-inline: ($spacing-xs-em * 0.5);
+  padding-block: calc(var(--SpacePaddingBlockDismiss) * 0.5);
```

Comment:
Should we / can we handle adjustments like this in the theme file? That way, `--SpacePaddingBlockDismiss` represents the value applied.

Ex: `SpacePaddingBlockDismiss: theme.SpaceEmXs * 2`

The use of a bespoke, tiny unit here is a design smell to me, so no worries if this is a hassle. I'd like to revisit this later with Design.

Reply:
@edburyenegren-okta I'm glad you brought this up. I agree all the little micro adjustments via `calc` or even other compile time only sass functions move us **away** from having immutable token values used consistently throughout our system.

I'm sure there will always be exceptions to our rules, but I think I'd prefer to see those clearly written as **static** values in a component theme file where I don't need to do math to see what the resulting value is.

Furthermore, simplified static values help because the component theme variables are **never** compiled. This is a very sharp edge case in our current design. If I try to create a component theme variable that looks like this: `ColorBackgroundDismissDisabled: "transparentize(white, 0.6)"` it will break our layout. There is no further SASS or PostCSS compilation happening after theme values are injected today. This problem because more pernicious when we consider cross browser CSS and everything PostCSS does to help us with IE11 at compile time. Keeping theme values simple will help us, but we can also explore solving this problem more thoroughly by compiling component theme variables (or even global tokens) the same way we do style modules.

Reply:
Mk, that makes sense. It's probably better that calculations like this (1) don't take place outside the style module (2) are utilized when they're programmatic rather than being used to achieve a magic number.

`SpacePaddingBlockDismiss: "0.1875em"` will give us the proper result here, but will become disconnected from our base spacing units. :/

I'm good with leaving this `calc()` in place so we know what the source of the proportion should be. We can either adjust this value to a standard or create a new spacing unit when we have time for a design review.